### PR TITLE
[opt](invert index) optimize the code structure of inverted index que…

### PR DESCRIPTION
**Pull request by zzzxl1993**
…ries

## Proposed changes

Issue Number: close #xxx

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: docs/zh-CN/docs/data-table/index/inverted-index.md [hunk 1]
```diff
@@ -50,7 +50,10 @@ under the License.
 Doris倒排索引的功能简要介绍如下：
 
 - 增加了字符串类型的全文检索
-  - 支持字符串全文检索，包括同时匹配多个关键字MATCH_ALL、匹配任意一个关键字MATCH_ANY、匹配短语词组MATCH_PHRASE
+  - 支持字符串全文检索，包括同时匹配多个关键字MATCH_ALL、匹配任意一个关键字MATCH_ANY
```
Comment (qidaye) [inline docs/zh-CN/docs/data-table/index/inverted-index.md:53]: Please update English version doc too.

File: be/src/olap/rowset/segment_v2/inverted_index/query/query.h [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+CL_NS_USE(index)
+CL_NS_USE(search)
+
+namespace doris::segment_v2 {
+
+class Query {
+public:
+    virtual ~Query() = default;
+
+    virtual void add(const std::wstring& field_name, const std::vector<std::string>& terms) = 0;
```
Comment (qidaye) [inline be/src/olap/rowset/segment_v2/inverted_index/query/query.h:40]: Add some comment to explain the function.

**Comment by zzzxl1993** [comment]
run buildall

**Review by qidaye — DISMISSED** [review]
LGTM

**Comment by zzzxl1993** [comment]
run compile

**Comment by zzzxl1993** [comment]
run compile

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run p0

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index/query/query_factory.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+class PhrasePrefixQuery;
+class RegexpQuery;
+
+class QueryFactory {
+public:
+    template <typename... Args>
+    static std::unique_ptr<Query> create(InvertedIndexQueryType query_type, Args&&... args) {
+        if (query_type == InvertedIndexQueryType::MATCH_ANY_QUERY ||
+            query_type == InvertedIndexQueryType::EQUAL_QUERY) {
+            return std::make_unique<DisjunctionQuery>(std::forward<Args>(args)...);
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/inverted_index/query/query_factory.h:37]: maybe switch case is more clear than if else？

File: be/src/olap/rowset/segment_v2/inverted_index/query/phrase_prefix_query.cpp [hunk 4]
```diff
@@ -17,47 +17,56 @@
...
-                _CLDECDELETE(t);
+                _CLLDECDELETE(t);
             }
         }
     }
 }
 
 void PhrasePrefixQuery::search(roaring::Roaring& roaring) {
-    _searcher->_search(&_query, [&roaring](const int32_t docid, const float_t /*score*/) {
+    _searcher->_search(_query, [&roaring](const int32_t docid, const float_t /*score*/) {
```
Comment (airborne12) [inline be/src/olap/rowset/segment_v2/inverted_index/query/phrase_prefix_query.cpp:67]: we can use std::unique_ptr for query, then here we can pass _query.get() to search params

**Comment by zzzxl1993** [comment]
run buildall

File: be/src/olap/rowset/segment_v2/inverted_index/query/query_factory.h (hunk 3 above)
Comment (zzzxl1993) [reply be/src/olap/rowset/segment_v2/inverted_index/query/query_factory.h:38]: Change to switch

File: be/src/olap/rowset/segment_v2/inverted_index/query/phrase_prefix_query.cpp (hunk 4 above)
Comment (zzzxl1993) [reply be/src/olap/rowset/segment_v2/inverted_index/query/phrase_prefix_query.cpp]: Change to unique_ptr

**Comment by zzzxl1993** [comment]
run buildall

**Comment by zzzxl1993** [comment]
run compile

**Comment by zzzxl1993** [comment]
run buildall

**Review by airborne12 — APPROVED** [review]
LGTM

**Review by qidaye — APPROVED** [review]
LGTM

**Pull request merged**